networkx/algorithms/dag.py

```diff
@@ -1290,20 +1294,19 @@ def compute_v_structures(G):
     Examples
     --------
     >>> G = nx.DiGraph()
-    >>> G.add_edges_from([(1, 2), (0, 5), (3, 1), (2, 4), (3, 1), (4, 5), (1, 5), (5, 2)])
-    >>> sorted(nx.compute_v_structures(G))
-    [(0, 5, 1), (0, 5, 4), (1, 5, 4)]
+    >>> G.add_edges_from([(1, 2), (0, 4), (2, 3), (3, 4), (1, 4), (4, 2)])
+    >>> vstrct = [(min(p1, p2), v, max(p1, p2)) for p1, v, p2 in v_structures(G)]
+    >>> sorted(vstrct)
```

Comment:
Yes -- I see these errors and they are not errors on the other tests. I thought the order of nodes would be determined by the order of the edges provided to nx.DiGraph.  What is different in this Dispatch test environment that reverses the node order? hmmm...   I'd like to investigate further. I'm more bothered by not understanding what is going on than by including sorting. So if I don't find anything we should just put the sorts back in.

Reply:
it's not just the dispatching test. The (ubuntu,3.11) test also failed(ref. 1st wf failure from the above comment). The dispatching test also uses the same config i.e. ubuntu-latest and python 3.11. So, it might not have anything to do with dispatching.

> yes using `sorted` is necessary, ref the doctest failures(from the recent github wf failure) below(ref. https://github.com/networkx/networkx/actions/runs/8687244560/job/23820377679):
> 
> ```
> =================================== FAILURES ===================================
> _________________ [doctest] networkx.algorithms.dag.colliders __________________
> 1245     ------
> 1246     NetworkXNotImplemented
> 1247         If `G` is an undirected graph.
> 1248 
> 1249     Examples
> 1250     --------
> 1251     >>> G = nx.DiGraph()
> 1252     >>> G.add_edges_from([(1, 2), (0, 4), (2, 3), (3, 4), (1, 4), (4, 2)])
> 1253     >>> colliders = [(min(p1, p2), v, max(p1, p2)) for p1, v, p2 in nx.colliders(G)]
> 1254     >>> colliders
> Expected:
>     [(0, 4, 1), (0, 4, 3), (1, 2, 4), (1, 4, 3)]
> Got:
>     [(1, 2, 4), (0, 4, 3), (0, 4, 1), (1, 4, 3)]
> 
> /home/runner/work/networkx/networkx/networkx/algorithms/dag.py:1254: DocTestFailure
> ________________ [doctest] networkx.algorithms.dag.v_structures ________________
> 1290     ------
> 1291     NetworkXNotImplemented
> 1292         If `G` is an undirected graph.
> 1293 
> 1294     Examples
> 1295     --------
> 1296     >>> G = nx.DiGraph()
> 1297     >>> G.add_edges_from([(1, 2), (0, 4), (2, 3), (3, 4), (1, 4), (4, 2)])
> 1298     >>> vstructures = [(min(p1, p2), v, max(p1, p2)) for p1, v, p2 in nx.v_structures(G)]
> 1299     >>> vstructures
> Expected:
>     [(0, 4, 1), (0, 4, 3), (1, 4, 3)]
> Got:
>     [(0, 4, 3), (0, 4, 1), (1, 4, 3)]
> 
> /home/runner/work/networkx/networkx/networkx/algorithms/dag.py:1299: DocTestFailure
> ```